File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int mw = 0;
        int mh = 0;

        for (int[] s : sizes) {
            if (s[0] > s[1]) {
                int temp = s[0];
                s[0] = s[1];
                s[1] = temp;
            }

            if (s[0] > mw) mw = s[0];
            if (s[1] > mh) mh = s[1];
        }
        return mw * mh;
    }
}

File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.kt
class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        var mw = 0
        var mh = 0
        
        sizes.forEach {
            if(it[0] > it[1]) {
                val temp = it[0]
                it[0] = it[1]
                it[1] = temp
            }
            
            if(it[0] > mw) mw = it[0]
            if(it[1] > mh) mh = it[1]
        }
        return mw * mh
    }
}